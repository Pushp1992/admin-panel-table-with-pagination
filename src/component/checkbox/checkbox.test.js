import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import CheckBox from './checkbox';
import '@testing-library/jest-dom';

afterEach(() => {
    cleanup();
});

it('matches snapshot with actual DOM tree', () => {
    const component = renderer.create(<CheckBox/>);
    const DOMTree = component.toJSON();
    expect(DOMTree).toMatchSnapshot();
});

it('renders Checkbox component', () => {
    render(<CheckBox/>);
    const element = screen.getAllByTestId('checkbox');
    expect(element[0]).toBeInTheDocument();
});
