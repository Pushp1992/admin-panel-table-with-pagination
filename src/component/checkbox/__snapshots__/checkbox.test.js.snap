// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with actual DOM tree 1`] = `
Array [
  <input
    data-testid="checkbox"
    type="checkbox"
  />,
  <label />,
]
`;
